### ISSUE TITLE
Implement case insensitive code completion and put 'self' as the first result on methods

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1340]
_From [ricardob...@gmail.com](https://code.google.com/u/116032747765426583751/) on 2013-04-04T10:27:56Z_

A few weeks ago I started my first PyQt4 project. Aside from the GUI I need to use numpy and scipy. For plotting I went with matplotlib.

So my first thought was using eclipse with PyDev plugin. But I didn't quite like it, so I went back to use spyder which I found it to be faster and more lightweight than eclipse. But i keep missing two PyDev feature for code completion. 

1. Code completion in PyDev is NOT case sensitive. This very useful in packages like PyQt4 where most object's names, methods and atributtes are camelcased, or start with a capital letter. In spyder code completion is case-sensitive, which becomes annoying when don't exactly remember where the capital letters should be. 

2. When coding a class in PyDev is you type and 's' the very first suggestion is self, which become very handy in saving you a lot of key stokes.

I throw this two suggestion to the spyder community, and hope that someone thinks this is useful. Off course, this is just an enhancement with low priority.

Once again terrific job you guys are doing here.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1340_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-04-11T11:05:06Z_

I think both are sensible enhancement proposals. IPython also works as PyDev so it'd be good to match Spyder with it too.

I'll try to work on them after 2.2 is released.

**Summary:** Implement case insensitive code completion and put 'self' as the first result on methods  \<span class="oldvalue"> (was: Case sensitivity for code completion and 'self') \</span>  
  **Labels:** -Type-Defect Type-Enh Cat-CodeCompletion MS-v2.2

**Comment by spyder-issues-migrator** [comment]
_From [jed.lud...@gmail.com](https://code.google.com/u/114333776555678317161/) on 2013-04-29T15:08:37Z_

**Labels:** MS-v2.2.1

**Comment by spyder-issues-migrator** [comment]
_From [jed.lud...@gmail.com](https://code.google.com/u/114333776555678317161/) on 2013-04-29T15:10:06Z_

**Labels:** -MS-v2.2

**Comment by spyder-issues-migrator** [comment]
_From [jed.lud...@gmail.com](https://code.google.com/u/114333776555678317161/) on 2013-04-30T05:12:46Z_

**Labels:** -MS-v2.2.1 MS-v2.3

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-05-11T06:00:34Z_

**Labels:** -MS-v2.3 MS-v2.4

**Comment by goanpeca** [comment]
@blink1073, you have been working with completion right? Tou think you could tackle this?

**Comment by blink1073** [comment]
As is stands we are only invoking completion after a `.` or if the user hits `TAB`.  This is related to #1914.  I'd say we need to add another option called something like "autocomplete while typing" that is invoked for `s` at the start of a word and after three characters are typed.

**Comment by goanpeca** [comment]
Yes you are right, I misread. Still, can you take a look at it eventually?

**Comment by blink1073** [comment]
Yep, self-assigned.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Will (or could) case-insensitive be implemented as part of (or as a followup to) #4751 ? This would be a big benefit to me as well, and one of the things (along with on the fly autocomplete, ala #3708 ) I really miss from Rstudio.

**Comment by ccordoba12** [comment]
I think this something different from #4751, but sure, we should fix it in Spyder 4.

**Comment by MuellerSeb** [comment]
At the moment code completion is case insensitive in Spyder 4.0.0b3. It would be nice having a switch to choose if completion should be case sensitive or not, since some people prefer the one over the other.
An annoying case comes up for me, when I want to  auto-complete ``import`` starting with:

    im<tab>

Then the first thing showing up, is not ``import`` but ``ImportError`` which could be prevented if auto-complete would be case-sensitive. Or did I miss something?
Cheers, Sebastian
